### PR TITLE
Refactor release flow and use `changeset` to manage versioning

### PR DESCRIPTION
**Pull request by TuvalSimha**
Things to do:

- [x] Install Changesets Bot: https://github.com/apps/changeset-bot
- [x] Repository Settings: Configure GitHub Actions permissions: Go to repo Settings > Actions > General and make sure to configure the following:

Actions permissions should be set to Allow all actions and reusable workflows
Workflow permissions should be set to Read and write permissions, and make sure the Allow GitHub Actions to create and approve pull requests option is active.

- [x] Unified secrets:
You can create an NPM publishing token by using npm token create.

After creating your token, make sure to add it as part of your GitHub Actions Secrets (under repo Settings). Name it NPM_TOKEN.

In addition, the shared pipelines are going to use GITHUB_TOKEN provided by GitHub Actions runtime. You can customize it by creating a custom PAT token for the user you wish to use.

### CODE REVIEW

**Review by Urigo — APPROVED** [review]

**Comment by TuvalSimha** [comment]
@dotansimha 
Should I remove the `md` file of `Changesets` for the first time, right?

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -9,7 +9,22 @@ jobs:
...
+      setupEnvWorkingDirectory: "./website"
+    secrets:
+      githubToken: ${{ secrets.GITHUB_TOKEN }}
+
+  release:
+    uses: the-guild-org/shared-config/.github/workflows/release-snapshot.yml@main
+    with:
+      npmTag: alpha
+      buildScript: build
+      nodeVersion: 18
```
Comment (dotansimha) [inline .github/workflows/pr.yml:22]: should be 16? the `release.yaml` is using 16... let's make sure it matches the other pipelines in SOFA repo

File: package.json [hunk 2]
```diff
@@ -42,6 +42,7 @@
     "graphql": "^15.0.0 || ^16.0.0"
   },
   "dependencies": {
+    "@changesets/cli": "2.26.0",
```
Comment (dotansimha) [inline package.json:45]: `@changesets/changelog-github` is missing?

File: package.json [hunk 3]
```diff
@@ -98,6 +97,10 @@
       "prettier --write"
     ]
   },
+  "publishConfig": {
+    "directory": "dist",
```
Comment (dotansimha) [inline package.json:101]: please double check that SOFA build actually created under `dist` directory

File: tsconfig.example.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "./tsconfig.json",
+  "extends": ["github>the-guild-org/shared-config:renovate", "./tsconfig.json"],
```
Comment (dotansimha) [inline tsconfig.example.json:2]: I think that's the wrong place for the renovate config? it should be in `renovate.json`

File: tsconfig.example.json [hunk 5]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "./tsconfig.json",
+  "extends": ["./renovate.json", "./tsconfig.json"],
```
Comment (dotansimha) [inline tsconfig.example.json:2]: that's not what I meant :) 
Remove all changes from this file, and then add the `extends` thing to `renovate.json`

**Comment by dotansimha** [comment]
@TuvalSimha please rebase, and do you know why we don't see the alpha release comment here?

**Comment by TuvalSimha** [comment]
@dotansimha  Fix the `Snapshot Release`, have a question about the SOFA repo because this repo is a mono-repo, I cant ignore the `website` in ignore option for `changeset`?

**Comment by dotansimha** [comment]
> @dotansimha Fix the `Snapshot Release`, have a question about the SOFA repo because this repo is a mono-repo, I cant ignore the `website` in ignore option for `changeset`?

You can. Either by adding `private: true` to the website `package.json`, or by adding `ignore: [PACKAGE_NAME]` to the changeset config file

**Review by dotansimha — APPROVED** [review]

**Pull request merged**